packages/stripe-adapter/test/adapter.test.ts

```diff
@@ -14,255 +14,204 @@
-      expect(product?.value).toEqual(expect.objectContaining({ id: 'prod_JMpOpTpdX5rDKx' }))
+    const PRODUCT_NAME = 'sfdsfsf_prod_JMpOpTpdX5rDKx'
+    it(`fetches prices for product "${PRODUCT_NAME}"`, () => {
+      const productInstance = <InstanceElement>fetchedInstances
+        .find(e => e.elemID.name === PRODUCT_NAME)
+      const expectedPriceIds = [
+        'price_1JjN8THipyrr1EYirgGmnx9A',
+        'price_1JjN8GHipyrr1EYi69M6nDJA',
+        'price_1Ik5jyHipyrr1EYiTX75FImK',
+        'price_1Ik5jyHipyrr1EYiZkn6nySi',
+      ]
+      const productPrices: Values[] = productInstance.value.product_prices
+      const receivedPriceIds: string[] = productPrices.map(p => p.id)
+      expect(receivedPriceIds).toIncludeSameMembers(expectedPriceIds)
+    })
 
-      const reportType = elements.filter(isInstanceElement).find(e => e.elemID.getFullName() === 'stripe.reporting_report_type.instance.balance_summary_1')
-      expect(reportType).toBeDefined()
-      expect(reportType?.value).toEqual(expect.objectContaining({ id: 'balance_summary_1' }))
+    describe('custom config', () => {
+      describe('fetches included types', () => {
+        const INCLUDE_TYPES = ['coupons', 'tax_rates']
+        const SINGULAR_INCLUDE_TYPES = INCLUDE_TYPES.map(t => pluralToSingularTypes[t])
 
+        const notIncluded = (typeName: string): boolean =>
+          !SINGULAR_INCLUDE_TYPES.includes(typeName)
 
-      const taxRate = elements.filter(isInstanceElement).find(e => e.elemID.getFullName() === 'stripe.tax_rate.instance.txr_1IcrqNHipyrr1EYiTpN0iuVj')
-      expect(taxRate).toBeDefined()
-      expect(taxRate?.value).toEqual(expect.objectContaining({ id: 'txr_1IcrqNHipyrr1EYiTpN0iuVj' }))
+        let fetchedInstancesTypes: Set<string>
 
+        beforeAll(async () => {
+          const config = new InstanceElement(
+            'config',
+            configType,
+            {
+              [FETCH_CONFIG]: {
+                includeTypes: INCLUDE_TYPES,
+              },
+              [API_DEFINITIONS_CONFIG]: DEFAULT_API_DEFINITIONS,
+            }
+          )
+          const instances = await fetchInstances(config)
+          fetchedInstancesTypes = new Set(instances.map(e => e.elemID.typeName))
+        })
+
+        it.each(SINGULAR_INCLUDE_TYPES)(
+          '%s',
+          includedType => {
+            expect(fetchedInstancesTypes).toContain(includedType)
+          }
+        )
 
-      const webhookEndpoint = elements.filter(isInstanceElement).find(e => e.elemID.getFullName() === 'stripe.webhook_endpoint.instance.we_1IcrkHHipyrr1EYiSGhlscV7')
-      expect(webhookEndpoint).toBeDefined()
-      expect(webhookEndpoint?.value).toEqual(expect.objectContaining({ id: 'we_1IcrkHHipyrr1EYiSGhlscV7' }))
-    })
-  })
+        it('doesn\'t fetch additional types', () => {
+          const additionalTypes: string[] = Array.from(fetchedInstancesTypes).filter(notIncluded)
+          expect(additionalTypes).toBeEmpty()
```

Comment:
I don't think we need so many variables for this test. It makes it harder to follow on the eyes:
```suggestion
          expect(Array.from(fetchedInstancesTypes).filter(type => !SINGULAR_INCLUDE_TYPES.includes(type))).toHaveLength(0)
```

Reply:
In general, I try to keep my expect().matcher() lines short, especially for error reporting and stacktrace.
However I do agree it's not fun to jump around the code for a quite simple filter, so moved the `notIncluded` into the test.